Geek-Collectors-Network/geek-collectors-network
# feat(api): user tags endpoints

### Summary
Add API endpoints to create, add, remove, get user `tags` (aka User Interests) and `userToTags`

Resolves #100 User interests endpoints

Add endpoints:

- [x] POST `/api/v1/user/tag/:tagId?` will create userToTag, or if no tagId and body contains `{ text: <tag text> }` create new tag and userToTag
- [x] DELETE `/api/v1/user/tag/:tagId` to delete the corresponding `usersToTags` entity (remove an interest)
- [x] GET `/api/v1/user/:userId?/tags` gets a list of all the give user's tags

### Retrospective
This took a while because there were other changes to schema and project structure that affected this. 

### Reviewer's Notes
~~For readability and simplicity I'm defining routes using POST and a descriptive endpoint rather than with DELETE etc.~~ refactored to use RESTful conventions.

I've gone with simply using the term *tags* over *interests* as it's more general and simplifies things. The front end will still display user tags as interests, and item tags as... whatever we decide to call them, if anything different.

### Testing
1. login with: alice.smith@email.com, password
2. GET `http://0.0.0.0:3000/api/v1/user/tags/` should return 3 tags, `http://0.0.0.0:3000/api/v1/user/4/tags/` returns Jane Doe's 4 tags
3. Create a new tag POST`http://0.0.0.0:3000/api/v1/user/tag/` with body {tag: Anime} - response contains `{ tagId, tagText }` and was added to db, also check `user_to_tag` table to see tag assoc. with user
4. POST `http://0.0.0.0:3000/api/v1/user/tag/10` will add the Pokemon tag in `user_to_tag`
5. DELETE `http://0.0.0.0:3000/api/v1/user/tag/10` will remove the Pokemon tag in `user_to_tag`
